tests/sweep_framework/sweeps/ccl/all_gather_n300.py

```diff
@@ -0,0 +1,136 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Optional, Tuple
+
+import torch
+
+import ttnn
+
+from tests.ttnn.utils_for_testing import check_with_pcc, start_measuring_time, stop_measuring_time
+from loguru import logger
+import ttnn
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
+from tests.ttnn.unit_tests.operations.test_all_gather import is_unsupported_case_n300
+from ttnn import ShardTensorToMesh
+
+# Override the default timeout in seconds for hang detection.
+TIMEOUT = 30
+
+# Parameters provided to the test vector generator are defined here.
+# They are defined as dict-type suites that contain the arguments to the run function as keys, and lists of possible inputs as values.
+# Each suite has a key name (in this case "suite_1" and "suite_2") which will associate the test vectors to this specific suite of inputs.
+# Developers can create their own generator functions and pass them to the parameters as inputs.
+
+batch_sizes = [1, 2]
+heights = [32, 64, 128]
+widths = [32 * (2**i) for i in range(10)]  # [32, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16384]
+input_shapes = []
+for W in batch_sizes:
+    for Z in batch_sizes:
+        for height in heights:
+            for width in widths:
+                input_shapes.append([W, Z, height, width])
+
+parameters = {
+    "all_gather_n300": {
+        "num_devices": [2],
+        "num_links": [1],
+        "input_shape": input_shapes,
+        "dim": [0, 1, 2, 3],
+        "layout": [ttnn.ROW_MAJOR_LAYOUT, ttnn.TILE_LAYOUT],
+        "input_dtype": [ttnn.bfloat16, ttnn.bfloat8_b],
+        "mem_config": [
+            ttnn.MemoryConfig(buffer_type=ttnn.BufferType.DRAM),
+            ttnn.MemoryConfig(buffer_type=ttnn.BufferType.L1),
+        ],
+        "enable_async": [True, False],
```

Comment:
for these sweeps we should only need to run in async mode. I don't think I mentioned this before (will also cut number of cases in half)